### PR TITLE
Support for IO constraints read from XDC

### PR DESCRIPTION
**Pull request by tmichalak**
This PR adds support for the LOC property in XDC and is required for the main [PR](https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1519) in symbiflow-arch-defs.
In most cases the LOC property is set on the IOBUF connected to the specified port.
For OBUFDS though there are two ports tracing to the same instance. In such case the LOC's parameter values are concatenated with a comma as separator, e.g. `"LOC": "ddram_dqs_p[0]:N2,ddram_dqs_n[0]:N1"`

A minilitex_ddr test has been added to test the LOC property settings.

### CODE REVIEW

File: xdc-plugin/tests/Makefile [hunk 1]
```diff
@@ -1,8 +1,14 @@
+# counter - basic test for IOSTANDARD, SLEW, DRIVE, IN_TERM properties
+# port_indexes - like counter but bus port indices are passes without curly braces
+# minilitex_ddr_arty - litex design with more types of IOBUFS including differential
+
 TESTS = counter \
```
Comment (mithro) [inline xdc-plugin/tests/Makefile:5]:
```suggestion
TESTS = \
    	counter \
    	port_indexes \
	minilitex_ddr_arty
```

File: xdc-plugin/tests/minilitex_ddr_arty.tcl [hunk 2]
```diff
@@ -0,0 +1,15 @@
+yosys -import
```
Comment (mithro) [inline xdc-plugin/tests/minilitex_ddr_arty.tcl:1]: Can you put this test in it's own directory

Comment (litghost) [reply xdc-plugin/tests/minilitex_ddr_arty.tcl:1]: Agreed

File: xdc-plugin/tests/minilitex_ddr_arty_golden.json [hunk 3]
```diff
@@ -0,0 +1,247 @@
+{
+  "IOBUF": {
+    "IN_TERM": "UNTUNED_SPLIT_40",
+    "IOSTANDARD": "SSTL135",
+    "LOC": "ddram_dq[0]:K5",
```
Comment (litghost) [inline xdc-plugin/tests/minilitex_ddr_arty_golden.json:5]: This is not the LOC syntax from Vivado.  Can we use a new parameter, e.g. XLOC or LOC_PAIRS, or IO_LOC, etc?

File: xdc-plugin/xdc.cc [hunk 4]
```diff
@@ -44,22 +44,23 @@ static bool isOutputPort(RTLIL::Wire* wire) {
 	return wire->port_output;
 }
 
-enum class SetPropertyOptions { INTERNAL_VREF, IOSTANDARD, SLEW, DRIVE, IN_TERM };
+enum class SetPropertyOptions { INTERNAL_VREF, IOSTANDARD, SLEW, DRIVE, IN_TERM, LOC };
```
Comment (litghost) [inline xdc-plugin/xdc.cc:47]: Please rename LOC to whatever the new parameter is going to be named

Comment (tmichalak) [reply xdc-plugin/xdc.cc]: Changed to `IO_LOC_PAIRS` as this is what it is.

File: xdc-plugin/tests/minilitex_ddr_arty_golden.json (hunk 3 above)
Comment (tmichalak) [reply xdc-plugin/tests/minilitex_ddr_arty_golden.json]: Done

File: xdc-plugin/tests/minilitex_ddr_arty.tcl (hunk 2 above)
Comment (tmichalak) [reply xdc-plugin/tests/minilitex_ddr_arty.tcl]: Sure, I moved all of the tests to separate directories.

File: xdc-plugin/tests/Makefile (hunk 1 above)
Comment (tmichalak) [reply xdc-plugin/tests/Makefile:6]: Added io_loc_pairs test.

**Pull request merged**